tools/amplicon_architect.cwl

```diff
@@ -0,0 +1,73 @@
+cwlVersion: v1.2
+class: CommandLineTool
+id: amplicon-architect
+doc: "This tool prepares input for amplicon architect anaylsis"
+requirements: 
+  - class: ShellCommandRequirement
+  - class: InlineJavascriptRequirement 
+  - class: DockerRequirement
+    dockerPull: 'jluebeck/prepareaa:latest'
+  - class: ResourceRequirement
+    ramMin: 16000
+    coresMin: $(inputs.threads)
+baseCommand: [mkdir, data_repo]
+arguments: 
+  - position: 1
+    shellQuote: false
+    valueFrom: >- 
+      && tar -C data_repo -xzf
+  - position: 2
+    shellQuote: false
+    valueFrom: >-
+      && export AA_DATA_REPO=$PWD/data_repo
+      && export AA_SRC=/home/programs/AmpliconArchitect-master/src
+      && mkdir licenses && cp $(inputs.mosek_license_file.path) licenses
+      && export MOSEKLM_LICENSE_FILE=$PWD/licenses
+      && cp $(inputs.coverage_stats.path) $AA_DATA_REPO/coverage.stats
+      ${
+        if (inputs.ref_cache){
+            var cmd = " && tar -xzf " + inputs.ref_cache.path
+            + " && export REF_CACHE=\"$PWD/ref/cache/%2s/%2s/%s\"";
+            return cmd;
+        }
+        else{
+            return "";
+        }
+      }
+  - position: 3
+    shellQuote: false
+    valueFrom: >-
+      && /home/programs/AmpliconArchitect-master/src/AmpliconArchitect.py
+inputs:
+  data_repo: { type: File, doc: "Reference tar ball obtained from https://datasets.genepattern.org/?prefix=data/module_support_files/AmpliconArchitect/", inputBinding: { position: 1} }
+  data_ref_version: { type: ['null', {type: enum, name: wgs_mode, symbols: ["GRCh38", "hg19", "GRCh37", "mm10", "GRCm38"]}], doc: "Genome version in data repo to use", default: "GRCh38", inputBinding: { position: 3, prefix: "--ref"} }
+  bam: { type: File, doc: "tumor bam file", secondaryFiles: ['^.bai?', '.crai?'], inputBinding: { position: 3, prefix: "--bam" } }
```

Comment:
These secondary file formats are not 1.2 compliant. You should use the pattern/required dict format.